lib/Pipes/TaggedFunctionKind.cpp

```diff
@@ -49,18 +55,223 @@ TaggedFunctionKind::symbolToTarget(const llvm::Function &Symbol) const {
-  const auto &CurrentModel = getModelFromContext(Ctx);
+                                const GlobalTupleTreeDiff &Diff,
+                                const pipeline::Global &Before,
+                                const pipeline::Global &After) const {
+  return;
+}
+
+// Find callers of a given `model::Function` via CrossRelations
+static void insertCallers(const TupleTree<CrossRelations> &CrossRelations,
+                          MetaAddress Entry,
+                          std::set<std::string> &Result) {
+  const auto Location = location(ranks::Function, Entry);
+  const auto &Related = CrossRelations->Relations.at(Location.toString())
+                          .Related;
+  for (const yield::RelationTarget &RT : Related) {
+    if (RT.Kind == yield::RelationType::IsCalledFrom) {
+      std::optional<MetaAddress>
+        Caller = genericLocationFromString<0>(RT.Location,
+                                              ranks::Function,
+                                              ranks::BasicBlock,
+                                              ranks::Instruction);
+      revng_assert(Caller.has_value());
+      Result.insert(Caller->toString());
+    }
+  }
+}
+
+// Find callers of a given `model::DynamicFunction` via CrossRelations
+static void insertCallers(const TupleTree<CrossRelations> &CrossRelations,
+                          std::string Entry,
+                          std::set<std::string> &Result) {
+  const auto Location = location(ranks::DynamicFunction, Entry);
+  const auto &Related = CrossRelations->Relations.at(Location.toString())
+                          .Related;
+  for (const yield::RelationTarget &RT : Related) {
+    if (RT.Kind == yield::RelationType::IsDynamicallyCalledFrom) {
+      std::optional<MetaAddress>
+        Caller = genericLocationFromString<0>(RT.Location,
+                                              ranks::Function,
+                                              ranks::BasicBlock,
+                                              ranks::Instruction);
+      revng_assert(Caller.has_value());
+      Result.insert(Caller->toString());
+    }
+  }
+}
+
+using IsolatedFK = IsolatedFunctionKind;
+void IsolatedFK::getInvalidations(const Context &Ctx,
+                                  TargetsList &ToRemove,
+                                  const GlobalTupleTreeDiff &Diff,
+                                  const pipeline::Global &Before,
+                                  const pipeline::Global &After) const {
+  using PM = PathMatcher;
+  using Binary = model::Binary;
+  if (not llvm::isa<TupleTreeGlobal<Binary>>(Before))
+    return;
+
+  const auto &OldModel = cast<TupleTreeGlobal<Binary>>(Before).get();
+  const auto &Model = cast<TupleTreeGlobal<Binary>>(After).get();
+
+  // If CrossRelations is not found, the most conservative approach is to
+  // invalidate all targets.
+  if (not Ctx.containsReadOnlyContainer(BinaryCrossRelationsRole)) {
+    ToRemove.emplace_back(Target(*this));
```

Comment:
Does this create a `*` target?
`*` is going to disappear soon, insert all the functions by hand.